x-pack/plugins/canvas/canvas_plugin_src/functions/common/palette.ts

```diff
@@ -52,15 +49,16 @@ export function palette(): ExpressionFunctionDefinition<'palette', null, Argumen
         options: [true, false],
       },
     },
-    fn: (input, args) => {
+    fn: (input, args, context) => {
       let colors: string[] = paulTor14.colors;
       const { color, reverse } = args;
       let { gradient } = args;
 
       if (color) {
         colors = color;
       } else {
-        const workpadPalette = getWorkpadPalette(getState());
+        const theme = context.variables.theme;
+        const workpadPalette = theme?.palette;
```

Comment:
This works and at least moves the redux state out of the function. Ideally though, we'd just set the call to the theme expression function in the default settings for the colors/gradient values.